src/main/java/org/ngafid/accounts/User.java

```diff
@@ -87,8 +87,15 @@ public boolean hasAggregateView() {
     /**
     * @return the user's email preferences
     */
-    public UserEmailPreferences getUserEmailPreferences() {
-        return userEmailPreferences;
+    public UserEmailPreferences getUserEmailPreferences(Connection connection) {
+
+        try {
+            return User.getUserEmailPreferences(connection, id);
+        } catch (SQLException e) {
+            e.printStackTrace();
+            return null;
+        }
+        
```

Comment:
This function should probably be declared to `throw SLQException` so it can propagate the error up to our other error handling code. Otherwise, we will just get a null pointer exception.